[Bug] Added Startup Probe in Quickstart Model

jolfr opened the pull request:
## Pull Request Description
Added startup probe to deal with slow cold-start model container in example model.

## Related Issues
Resolves: #772 

**Important: Before submitting, please complete the description above and review the checklist below.**

---

<details>
<summary><strong>Contribution Guidelines (Expand for Details)</strong></summary>

<p>We appreciate your contribution to aibrix! To ensure a smooth review process and maintain high code quality, please adhere to the following guidelines:</p>

<h3>Pull Request Title Format</h3>
<p>Your PR title should start with one of these prefixes to indicate the nature of the change:</p>
<ul>
    <li><code>[Bug]</code>: Corrections to existing functionality</li>
    <li><code>[CI]</code>: Changes to build process or CI pipeline</li>
    <li><code>[Docs]</code>: Updates or additions to documentation</li>
    <li><code>[API]</code>: Modifications to aibrix's API or interface</li>
    <li><code>[CLI]</code>: Changes or additions to the Command Line Interface</li>
    <li><code>[Misc]</code>: For changes not covered above (use sparingly)</li>
</ul>
<p><em>Note: For changes spanning multiple categories, use multiple prefixes in order of importance.</em></p>

<h3>Submission Checklist</h3>
<ul>
    <li>[ ] PR title includes appropriate prefix(es)</li>
    <li>[ ] Changes are clearly explained in the PR description</li>
    <li>[ ] New and existing tests pass successfully</li>
    <li>[ ] Code adheres to project style and best practices</li>
    <li>[ ] Documentation updated to reflect changes (if applicable)</li>
    <li>[ ] Thorough testing completed, no regressions introduced</li>
</ul>

<p>By submitting this PR, you confirm that you've read these guidelines and your changes align with the project's contribution standards.</p>

</details>

kerthcet commented on samples/quickstart/model.yaml:
```diff
@@ -51,7 +51,7 @@ spec:
               port: 8000
               scheme: HTTP
-            initialDelaySeconds: 120
+            initialDelaySeconds: 240
```
This is tricky because the waiting time depends on the model size and network heavily. Maybe quick start with a smaller model makes more sense,  like opt-125m.

jolfr replied:
Yeah reducing the model size is one way to go, it would guarantee quicker startup times. Another way could be increasing the failure threshold, that way we get quick startup times if the network is fast, but don't get pod restarts if it is slow.

jolfr replied:
Should have included machine specs in my original bug ticket. Running on a `g2-standard-4` from GCP, which has a max bandwidth of 10Gb/s, the slowest of the GPU machines, so model download is worst-case.

jolfr replied:
@kerthcet thoughts?

Jeffwan replied:
https://kubernetes.io/docs/concepts/configuration/liveness-readiness-startup-probes/#startup-probe @jolfr @kerthcet  WDYT?

kerthcet replied:
make sense to me

Jeffwan replied:
@jolfr could you make a change?

jolfr commented:
Two changes to original PR:

1. Instead of increasing static initialDelay to larger number, added startupProbe block, which will allow for dynamic initial delay. Set startupProbe to 5 minutes timeout.
2. Removed initialDelay from probes as startupProbe will now be handling the delay.

jolfr commented:
Amended the PR, ready for you guys to review 👍🏻

Jeffwan reviewed (commented):
The change looks good to me. I think @kerthcet agreed on introducing the startupProbe. We can merge this change now.
